benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+    """
+    genes = {}
+    if verbose:
+        sys.stdout.write("\tParse GenBank record ...\n")
+    seq = Sequence.read(genbank_fp, format='genbank')
+    if verbose:
+        sys.stdout.write("\t\tDone.\n")
+    for feature in list(seq.interval_metadata.features.keys()):
+        if feature['type_'] == 'CDS':
+            protein_id = feature['protein_id']
+            translation = feature['translation']
+            strand = '+'
+            if feature['rc_']:
+                strand = '-'
+            loc = seq.interval_metadata.features[feature]
+            start_pos = loc[0][0]
+            end_pos = loc[0][1]
+            if protein_id not in genes:
+                genes[protein_id.replace("\"", "")] = [
+                    translation.replace(" ", "").replace("\"", ""),
+                    start_pos, end_pos, strand]
+            else:
+                raise KeyError("%s already exists in dictionary" % protein_id)
+    return seq, genes
+
+
+def launch_orthofinder(proteomes_dir, threads, verbose=False):
+    """Launch OrthoFinder to report orthologous gene groups for two genomes.
+
+    Parameters
+    ----------
+    proteomes_dir: string
+        directory path storing FASTA coding sequences for complete genomes
+    threads: integer
+        number of threads to use
+    verbose: boolean
+        if True, run in verbose mode
+    """
+    if verbose:
+        sys.stdout.write("\tLaunch OrthoFinder ..\n")
+    orthofinder_command = ["orthofinder.py",
+                           "-f", proteomes_dir,
+                           "-t", str(threads)]
+    proc = subprocess.Popen(orthofinder_command,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            close_fds=True)
+    proc.wait()
+    stdout, stderr = proc.communicate()
+    if stderr:
+        print(stderr)
+    if verbose:
+        print(stdout)
+        sys.stdout.write("\tDone\n")
+
+
+def parse_orthofinder(results_dir):
+    """Parse the output files of OrthoFinder for orthologous genes.
+
+    Parameters
+    ----------
+    results_dir: string
+        OrthoFinder results directory
+
+    Returns
+    -------
+    species_ids: dictionary
+    sequence_ids: dictionary
+    orthologous_groups: list of lists
+    """
+    species_ids = {}
+    sequence_ids = {}
+    orthologous_groups = []
+    with open(
+        glob.glob(
+            join(
+                results_dir,
+                "clusters_OrthoFinder_*_id_pairs.txt"))[0], 'r') as id_pairs_f:
+        # skip header lines
+        for _ in range(7):
+            next(id_pairs_f)
+        # parse orthologous family groups
+        for line in id_pairs_f:
+            line = line.strip().split()
+            # include only families with at least 2 orthologs
+            if len(line[1:-1]) > 1:
+                orthologous_groups.append(line[1:-1])
+    with open(join(results_dir, "SpeciesIDs.txt"), 'r') as species_ids_f:
```

Comment:
it looks like the logic for parsing species IDs and sequence IDs is the same, would it make sense to decompose context block? eg

```python
species_ids = _parse_orthofinder_ids(join(results_dir, 'SpeciesIDs.txt'))
sequence_ids = _parse_orthofinder_ids(join(results_dir, 'SequenceIDs.txt'))
```